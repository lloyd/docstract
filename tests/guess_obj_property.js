// assignment to an object property should be name-guessable, this test
// has two examples.

  /**
   * Closes both the stream and its backing stream.  If the stream is already
   * closed, an exception is thrown.  For TextWriters, this first flushes the
   * backing stream's buffer.
   */
  stream.close = function stream_close() {
    self.ensureOpened();
    self.unload();
  };

  /**
   * Writes a string to the stream.  If the stream is closed, an exception is
   * thrown.
   *
   * @param str
   *        The string to write.
   */
  this.write = function TextWriter_write(str) {
    manager.ensureOpened();
    let istream = uconv.convertToInputStream(str);
    let len = istream.available();
    while (len > 0) {
      stream.writeFrom(istream, len);
      len = istream.available();
    }
    istream.close();
  };
