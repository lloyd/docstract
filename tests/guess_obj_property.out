{
  "filename": "tests/guess_obj_property.js", 
  "functions": [
    {
      "desc": "Closes both the stream and its backing stream.  If the stream is already\nclosed, an exception is thrown.  For TextWriters, this first flushes the\nbacking stream's buffer.", 
      "name": "close", 
      "source_lines": [
        4, 
        8
      ]
    }, 
    {
      "desc": "Writes a string to the stream.  If the stream is closed, an exception is\nthrown.", 
      "name": "write", 
      "params": [
        {
          "desc": "The string to write.", 
          "name": "str"
        }
      ], 
      "source_lines": [
        14, 
        20
      ]
    }
  ], 
  "module": "guess_obj_property"
}
